blob/s3blob/s3blob.go

```diff
@@ -670,12 +667,17 @@ func (b *bucket) NewTypedWriter(ctx context.Context, key string, contentType str
 	}, nil
 }
 
-var errUnimplemented = errors.New("not implemented")
-
 // Copy implements driver.Copy.
 func (b *bucket) Copy(ctx context.Context, dstKey, srcKey string, opts *driver.CopyOptions) error {
-	// TODO(rvangent): Implement this and delete errUnimplemented.
-	return errUnimplemented
+	dstKey = escapeKey(dstKey)
+	srcKey = escapeKey(srcKey)
+	input := &s3.CopyObjectInput{
+		Bucket:     aws.String(b.name),
+		CopySource: aws.String(srcKey),
```

Comment:
According to the godoc, you need the bucket name as a prefix here, with a `/` separator.

`CopySource: aws.String(b.name + "/" + srcKey)`